src/cdk/testing/harness-environment.ts

```diff
@@ -119,6 +121,24 @@ export abstract class HarnessEnvironment<E> implements HarnessLoader, LocatorFac
     return (await this.getAllRawElements(selector)).map(e => this.createEnvironment(e));
   }
 
+  protected isCDBatching() {
+    return isCDBatching;
+  }
+
+  async batchCD<T>(fn: () => Promise<T>) {
+    const alreadyBatching = isCDBatching;
+    if (!alreadyBatching) {
+      await this.forceStabilize();
```

Comment:
Can you add a comment that explains why you force stabilize before batching changes?